tsdb/head.go

```diff
@@ -1653,6 +1653,16 @@ func (h *Head) NumStaleSeries() uint64 {
 	return h.numStaleSeries.Load()
 }
 
+// GetMetadataByRef returns the metadata for the series with the given reference.
+// Also returns the series labels hash for verification against ref reuse.
+func (h *Head) GetMetadataByRef(ref chunks.HeadSeriesRef) (meta *metadata.Metadata, labelsHash uint64, err error) {
+	series := h.series.getByID(ref)
+	if series == nil {
+		return nil, 0, nil
+	}
+	return series.meta, series.lset.Hash(), nil
```

Comment:
I think we need to lock the series `series.Lock()`, copy the metadata and unlock the series.
Something like:
```
series.Lock()
defer series.Unlock()
meta = *series.meta
```

Which also means that the return value should be just `meta metadata.Metadata`, not a pointer.